ios/Classes/SwiftOptimoveFlutterSdkPlugin.swift

```diff
@@ -0,0 +1,366 @@
+    
+    fileprivate func inAppGetInboxItems(_ result: @escaping FlutterResult) {
+        let inboxItems: [InAppInboxItem] = OptimoveInApp.getInboxItems()
+        
+        var inboxItemsMaps: [[String: Any?]] = []
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+        
+        inboxItems.forEach { item in
+            var inboxItemMap = [String: Any?]()
+            
+            inboxItemMap["id"] = item.id
+            inboxItemMap["title"] = item.title
+            inboxItemMap["subtitle"] = item.subtitle
+            inboxItemMap["sentAt"] = dateFormatter.string(from:  item.sentAt)
+            inboxItemMap["availableFrom"] = item.availableFrom != nil ? dateFormatter.string(from:  item.availableFrom!) : nil
+            inboxItemMap["availableTo"] = item.availableTo != nil ? dateFormatter.string(from:  item.availableTo!) : nil
+            inboxItemMap["dismissedAt"] = item.dismissedAt != nil ? dateFormatter.string(from:  item.dismissedAt!) : nil
+            inboxItemMap["isRead"] = item.isRead()
+            inboxItemMap["imageUrl"] = item.getImageUrl()?.absoluteString
+            inboxItemMap["data"] = item.data
+            
+            inboxItemsMaps.append(inboxItemMap)
+        }
+        
+        result(inboxItemsMaps)
+    }
+    
+    fileprivate func inAppDeleteMessageFromInbox(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
+        let id = (call.arguments as! Dictionary<String, Any>)["id"] as! Int
+        
+        let inboxItems: [InAppInboxItem] = OptimoveInApp.getInboxItems()
+        var deleted = false
+        inboxItems.forEach { item in
+            if item.id == id {
+                deleted = OptimoveInApp.deleteMessageFromInbox(item: item)
+            }
+        }
+        
+        result(deleted)
+    }
+    
+    fileprivate func handleInAppInboxSummary(_ result: @escaping FlutterResult) {
+        OptimoveInApp.getInboxSummaryAsync(inboxSummaryBlock: { inAppInboxSummary in
+            var inAppInboxSummaryMap = [String: Any?]()
+            inAppInboxSummaryMap["totalCount"] = inAppInboxSummary?.totalCount
+            inAppInboxSummaryMap["unreadCount"] = inAppInboxSummary?.unreadCount
+            result(inAppInboxSummaryMap)
+        })
+    }
+    
+    
+    fileprivate func emitPushNotificationReceivedEvent(pushNotification: PushNotification){
+        var notificationMap = [String: Any?]()
+        notificationMap["type"] = EventTypes.pushReceived.rawValue
+        notificationMap["data"] = getPushNotificatioMap(from: pushNotification)
+        self.eventSink?(notificationMap)
+    }
+    
+    fileprivate func emitPushNotificationOpenedEvent(pushNotification: PushNotification){
+        var notificationMap = [String: Any?]()
+        notificationMap["type"] = EventTypes.pushOpened.rawValue
+        notificationMap["data"] = getPushNotificatioMap(from: pushNotification)
+        self.eventSink?(notificationMap)
```

Comment:
Do we need to queue this in the case the sink isn't ready yet? (i.e. "cold" app starts from notification tap)

Reply:
Changed to a queue